web/client/components/misc/Popover.jsx

```diff
@@ -0,0 +1,320 @@
+                        && clientX <= left + width
+                        && clientY >= top
+                        && clientY <= top + height);
+                if (isOutsidePopover) {
+                    onOpen(false);
+                }
+            }
+        }
+        window.addEventListener('click', computeActive, true);
+        window.addEventListener('wheel', computeActive, true);
+        window.addEventListener('scroll', computeActive, true);
+        return () => {
+            window.removeEventListener('click', computeActive, true);
+            window.removeEventListener('wheel', computeActive, true);
+            window.removeEventListener('scroll', computeActive, true);
+        };
+    }, [ ]);
+
+    return (
+        <>
+        <div
+            className="ms-popover"
+            ref={swatch}>
+            {cloneElement(children, {
+                onClick: (event) => {
+                    event.stopPropagation();
+                    const newOpen = !open;
+                    onOpen(newOpen);
+                }
+            })}
+        </div>
+        {containerNode && open
+            ? createPortal(
+                <div
+                    className="ms-popover-overlay"
+                    ref={overlay}
+                    style={{
+                        position: 'fixed',
+                        display: 'flex',
+                        alignItems: 'center',
+                        justifyContent: 'center',
+                        width: '100%',
+                        height: '100%',
+                        top: 0,
+                        left: 0,
+                        pointerEvents: 'none',
+                        ...styles?.overlay
+                    }}>
+                    <div
+                        style={{
+                            position: 'absolute',
+                            width: '100%',
+                            height: '100%',
+                            top: 0,
+                            left: 0
+                        }}
+                    />
+                    <div
+                        ref={popover}
+                        style={{
+                            pointerEvents: 'auto',
+                            ...styles?.picker
+                        }}>
+                        {content}
+                    </div>
+                    <div
+                        style={{
+                            position: 'absolute',
+                            borderTop: `${margin - 1}px solid transparent`,
+                            borderBottom: `${margin - 1}px solid transparent`,
+                            borderRight: `${margin - 1}px solid #ffffff`,
+                            filter: 'drop-shadow(-4px 2px 4px rgba(0, 0, 0, 0.2))',
+                            ...styles?.arrow
+                        }}/>
+                </div>, containerNode)
+            : null }
+        </>
+    );
+}
+
+function Popover({
```

Comment:
Can you document this share d component, at least the purpose and the main props?  How it differs for other popover around the application?